buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-Personal

```diff
@@ -0,0 +1,361 @@
+
+            checkout scm
+            def variableFile = load 'buildenv/jenkins/common/variables-functions'
+            def buildFile = load 'buildenv/jenkins/common/pipeline-functions'
+
+            // parse variables file and initialize variables
+            variableFile.parse_variables_file()
+            USER_CREDENTIALS_ID = variableFile.get_git_user_credentials_id()
+
+            // set OpenJ9, OMR repos, branches and SHAs
+            variableFile.set_extensions_variables(EXTENSIONS)
+
+            if (!OPENJ9_SHA) {
+                OPENJ9_SHA = buildFile.get_repository_sha(OPENJ9_REPO, OPENJ9_BRANCH)
+            }
+
+            if (!OMR_SHA) {
+                OMR_SHA = buildFile.get_repository_sha(OMR_REPO, OMR_BRANCH)
+            }
+
+            // Update build name
+            wrap([$class: 'BuildUser']) {
+                currentBuild.displayName = "#${BUILD_NUMBER} - ${BUILD_USER_EMAIL}"
+            }
+
+            // update build description
+            currentBuild.description = params.PLATFORMS
+            currentBuild.description += "<br>OpenJ9: "
+            currentBuild.description += get_short_sha(OPENJ9_SHA)
+            currentBuild.description += "<br>OMR: "
+            currentBuild.description += get_short_sha(OMR_SHA)
+
+            SDK_VERSIONS.each { SDK_VERSION ->
+                def OPENJDK_REPO
+                def OPENJDK_BRANCH
+                def OPENJDK_SHA
+
+                params.each { key, value ->
+                    if (key.startsWith("OPENJDK${SDK_VERSION}_")) {
+                        switch (key) {
+                            case "OPENJDK${SDK_VERSION}_REPO":
+                                OPENJDK_REPO = value
+                                break
+                            case "OPENJDK${SDK_VERSION}_BRANCH":
+                                OPENJDK_BRANCH = value
+                                break
+                            case "OPENJDK${SDK_VERSION}_SHA":
+                                OPENJDK_SHA = value
+                                break
+                        }
+                    }
+                }
+
+                def OPENJDK_INFO = variableFile.get_openjdk_info(SDK_VERSION)
+                if (!OPENJDK_REPO) {
+                   // set it to the value set into the variable file
+                   OPENJDK_REPO = OPENJDK_INFO.get('repo')
+                }
+                if (!OPENJDK_BRANCH) {
+                   // set it to the value set into the variable file
+                   OPENJDK_BRANCH = OPENJDK_INFO.get('branch')
+                }
+                if (!OPENJDK_SHA){
+                    OPENJDK_SHA = buildFile.get_repository_sha(OPENJDK_REPO, OPENJDK_BRANCH)
+                }
+
+                // append OpenJDK SHA to the build description
+                currentBuild.description += "<br>OpenJDK${SDK_VERSION}: "
+                currentBuild.description += get_short_sha(OPENJDK_SHA)
```

Comment:
I'm not sure this will work. `get_short_sha` runs `git rev-parse` on the input `SHA`. Sounds like you would rather have a truncate here. Or better yet, We should change the get_sha function to return a short sha instead of the entire 40 char.

Reply:
It works. Should move the function to a common script to be called in other scripts too.

Reply:
It seems it works if you run it in any git repo. The sha doesn't have to be part of the repo you are in. Are we always guaranteed to have a .git folder in the dir where we run this?

Reply:
I think this only works if you supply a 40char (full) sha. if the user requests a sha and it is already shorter than 40 chars, this will fail

Reply:
It's always 40.
Might replace `git rev-parse` with groovy substring() or take() methods